src/ameba/rule/lint/literal_assignments_in_expressions.cr

```diff
@@ -0,0 +1,64 @@
+module Ameba::Rule::Lint
+  # A rule that disallows assignments with literal values
+  # in control expressions.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # if foo = 42
+  #   do_something
+  # end
+  # ```
+  #
+  # And should be replaced by the following:
+  #
+  # ```
+  # if foo == 42
+  #   do_something
+  # end
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Lint/LiteralAssignmentsInExpressions:
+  #   Enabled: true
+  # ```
+  class LiteralAssignmentsInExpressions < Base
+    properties do
+      description "Disallows assignments with literal values in control expressions"
+    end
+
+    MSG = "Detected assignment with a literal value in control expression"
+
+    PRIMITIVE_LITERAL_TYPES = {
+      Crystal::NilLiteral,
+      Crystal::BoolLiteral,
+      Crystal::NumberLiteral,
+      Crystal::CharLiteral,
+      Crystal::StringLiteral,
+      Crystal::SymbolLiteral,
+      Crystal::ProcLiteral,
+      Crystal::Path,
+    }
+
+    DYNAMIC_LITERAL_TYPES = {
+      Crystal::RangeLiteral,
+      Crystal::RegexLiteral,
+      Crystal::TupleLiteral,
+      Crystal::NamedTupleLiteral,
+      Crystal::ArrayLiteral,
+      Crystal::HashLiteral,
+    }
```

Comment:
can these constants be combined with existing methods? Maybe they could be moved to a single utility/file? 

https://github.com/crystal-ameba/ameba/blob/482597fbf4b5cb18c77aafb634b0f96eca585453/src/ameba/ast/util.cr#L3-L28

Reply:
I was thinking about that too, will do.

Reply:
I'll rebase this on top #304, once it lands.